tests/ui/rfcs/rfc-1623-static/rfc1623-2.rs

```diff
@@ -28,6 +28,10 @@ static SOME_STRUCT: &SomeStruct = &SomeStruct {
     f: &id,
     //~^ ERROR mismatched types
     //~| ERROR mismatched types
+    //~| ERROR mismatched types
+    //~| ERROR mismatched types
+    //~| ERROR implementation of `FnOnce` is not general enough
+    //~| ERROR implementation of `FnOnce` is not general enough
```

Comment:
I think the reason we get these more often is that now the promoteds created here are in required_const, so they get evaluated, and it's their evaluation that shows these errors.

This won't be visible on stable as it gets de-duplicated.